runtime/src/accounts_index.rs

```diff
@@ -94,13 +94,22 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
 
     // find the latest slot and T in a slice for a given ancestor
     // returns index into 'slice' if found, None if not.
-    fn latest_slot(&self, ancestors: Option<&Ancestors>, slice: SlotSlice<T>) -> Option<usize> {
+    fn latest_slot(
+        &self,
+        ancestors: Option<&Ancestors>,
+        slice: SlotSlice<T>,
+        max_root: Option<Slot>,
+    ) -> Option<usize> {
         let mut max = 0;
         let mut rv = None;
         for (i, (slot, _t)) in slice.iter().rev().enumerate() {
             if *slot >= max
                 && (ancestors.map_or(false, |ancestors| ancestors.contains_key(slot))
-                    || self.is_root(*slot))
+                    || (self.is_root(*slot)
```

Comment:
probably, extract this into `self.is_root_below_max()` or something similar. This condition is important but it's got a bit hard to read at first glance.

Reply:
extracted into `is_slot_rooted_and_le_max`